cpp/src/table/table_device_view.cu

```diff
@@ -52,5 +54,36 @@ template class table_device_view_base<column_device_view, table_view>;
 // Explicit instantiation for a device table of mutable views
 template class table_device_view_base<mutable_column_device_view, mutable_table_view>;
 
+namespace {
+struct is_relationally_comparable_impl {
+  template <typename T>
+  constexpr bool operator()()
+  {
+    return cudf::is_relationally_comparable<T, T>();
+  }
+};
+}  // namespace
+
+template <typename TableDeviceView>
+bool is_relationally_comparable(TableDeviceView const& lhs, TableDeviceView const& rhs)
+{
+  return thrust::all_of(thrust::counting_iterator<size_type>(0),
+                        thrust::counting_iterator<size_type>(lhs.num_columns()),
+                        [lhs, rhs] __device__(auto const i) {
+                          // Simplified this for compile time. (Ideally use double_type_dispatcher)
+                          // TODO: possible to implement without double type dispatcher.
+                          return lhs.column(i).type() == rhs.column(i).type() and
+                                 type_dispatcher(lhs.column(i).type(),
+                                                 is_relationally_comparable_impl{});
+                        });
+}
+// Explicit extern template instantiation for a device table of immutable views
```

Comment:
Why only device views? Shouldn't this be available for regular `table_view`?

Reply:
done.